packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -68,14 +68,63 @@ class ImageInfo {
 /// same stack frame as the call to [ImageStream.addListener]).
 typedef ImageListener = void Function(ImageInfo image, bool synchronousCall);
 
+/// Signature for listening to [ImageChunkEvent] events.
+///
+/// Used by [ImageStream].
+typedef ImageChunkListener = void Function(ImageChunkEvent event);
+
 /// Signature for reporting errors when resolving images.
 ///
 /// Used by [ImageStream] and [precacheImage] to report errors.
 typedef ImageErrorListener = void Function(dynamic exception, StackTrace stackTrace);
 
-class _ImageListenerPair {
-  _ImageListenerPair(this.listener, this.errorListener);
+/// An immutable notification of image bytes that have been incrementally loaded.
+///
+/// Chunk events represent progress notifications while an image is being
+/// loaded (e.g. from disk or over the network).
+///
+/// See also:
+///
+///  * [ImageChunkListener], the means by which callers get notified of
+///    these events.
+///  * [Image.imageLoadingBuilder], which allows callers to visually represent
+///    the loading progress of an image at the widget layer.
+class ImageChunkEvent extends Diagnosticable {
+  /// Creates a new chunk event.
+  const ImageChunkEvent(this.cumulativeBytesLoaded, this.expectedTotalBytes);
+
+  /// The number of bytes that have been received across the wire thus far.
+  final int cumulativeBytesLoaded;
+
+  /// The expected number of bytes that need to be received to finish loading
+  /// the image.
+  ///
+  /// This value is not necessarily equal to the expected _size_ of the image
+  /// in bytes, as the bytes required to load the image may be compressed.
+  ///
+  /// This value will be -1 if the number is not known in advance. When this is
```

Comment:
why -1 rather than null?

Reply:
'cause it's a passthrough from the `response.contentLength` property, which uses -1.  I could certainly see the argument for making this be null - wdyt?

Reply:
ah if it's just passing through an existing API, sounds good to me.

Reply:
if you haven't already, please reference the api you're forwarding in the docs